Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.cpp

```diff
@@ -54,10 +54,12 @@ using namespace WebCore;
 WTF_MAKE_TZONE_ALLOCATED_IMPL(RemoteImageBufferProxy);
 WTF_MAKE_TZONE_ALLOCATED_IMPL(RemoteSerializedImageBufferProxy);
 
-RemoteImageBufferProxy::RemoteImageBufferProxy(Parameters parameters, const ImageBufferBackend::Info& info, RemoteRenderingBackendProxy& remoteRenderingBackendProxy, std::unique_ptr<ImageBufferBackend>&& backend, RenderingResourceIdentifier identifier)
-    : ImageBuffer(parameters, info, { }, WTFMove(backend), identifier)
+RemoteImageBufferProxy::RemoteImageBufferProxy(Parameters parameters, const ImageBufferBackend::Info& info, RemoteRenderingBackendProxy& remoteRenderingBackendProxy, std::unique_ptr<ImageBufferBackend>&& backend, RemoteSerializedImageBufferReferenceTracker renderingResourceReferenceTracker, bool needsFlush)
+    : ImageBuffer(parameters, info, { }, WTFMove(backend), renderingResourceReferenceTracker.identifier())
+    , m_renderingResourceReferenceTracker(RemoteSerializedImageBufferReferenceTracker(renderingResourceReferenceTracker.add()))
     , m_remoteRenderingBackendProxy(remoteRenderingBackendProxy)
     , m_remoteDisplayList(*this, remoteRenderingBackendProxy, { { }, ImageBuffer::logicalSize() }, ImageBuffer::baseTransform())
+    , m_needsFlush(needsFlush)
```

Comment:
When a ImageBuffer is created, it doesn't have a context.
For us, flush is a conceptual operation that waits until draws done through a graphics context have been issued so that they're observable from the underlying surface.

It does not make sense that we would start ImageBuffer with "needsFlush == true", when by definition it doesn't have a context and as such can not have any commands to flush.

Reply:
`m_needsFlush` is already initialised to true unconditionally in the header.

Creating an ImageBuffer has an implicit clear operation to zero the previously undefined pixel contents, which results in a context being created immediately.

This clear isn't flushed though (since a blocking flush immediately after creating every image buffer is slow), so the initial state of the proxy is that it will need a flush before being read from.